Modules/_datetimemodule.c

```diff
@@ -11,14 +16,21 @@
 #  include <winsock2.h>         /* struct timeval */
 #endif
 
-/* Differentiate between building the core module and building extension
- * modules.
- */
-#ifndef Py_BUILD_CORE
-#define Py_BUILD_CORE
-#endif
-#include "datetime.h"
-#undef Py_BUILD_CORE
```

Comment:
I think it makes no difference in this case, but it's somewhat interesting that in the old version, `Py_BUILD_CORE` was un-defined regardless of whether it was originally defined.

I think this means that if `Py_BUILD_CORE` is defined when compiling `_datetimemodules.c`, it will now remain so for the rest of the file, as opposed to the pre-PR version, where it is unconditionally flipped off after importing `datetime.h`. I don't see any other references to it in the file, though, so it should be safe to remove this.